logstash_async/formatter.py

```diff
@@ -98,6 +98,10 @@ def format(self, record):
             'pid': record.process,
             'program': self._program_name,
             'type': self._message_type,
+            '@metadata': {
+                'beat': self._message_type,
+                'version': '1',
```

Comment:
I think it would be better to **not** hardcode the `beat` and `version` attributes.
Your current use case is to submit the common "Beat" metadata fields, but it can also be anything else what users want to transfer in the `@metadata` field. So, the content of the `@metadata` field should be passed in from the caller.
`python-logstash-async` should not know anything Beats.

Reply:
I have added the keyword argument `metadata`.

How do you want to deal with the lint issue: `logstash_async/formatter.py:29: [R0913(too-many-arguments), LogstashFormatter.__init__] Too many arguments (8/7)` ?

Reply:
Thanks for the changes.

Regarding the lint warning: I suggest ignoring it, like so:
```diff
@@ -26,6 +26,7 @@ except ImportError:
 class LogstashFormatter(logging.Formatter):
 
     # ----------------------------------------------------------------------
+    # pylint: disable=too-many-arguments
     def __init__(
             self,
             message_type='python-logstash',
```
Not perfectly nice but IMO in this case it is OK.

Reply:
Updated commits to resolve this